pkg/provider/provider.go

```diff
@@ -18,14 +26,68 @@ func Provider() *schema.Provider {
 				Type:        schema.TypeString,
 				Required:    true,
 				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_USER", nil),
+				// TODO validation
+			},
+			"password": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_PASSWORD", nil),
+				Sensitive:   true,
+			},
+			"role": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_ROLE", nil),
 			},
 			"region": &schema.Schema{
 				Type:        schema.TypeString,
 				Required:    true,
 				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_REGION", "us-west-2"),
 			},
 		},
-		ResourcesMap:   map[string]*schema.Resource{},
+		ResourcesMap: map[string]*schema.Resource{
+			"snowflake_database": resources.Database(),
+		},
 		DataSourcesMap: map[string]*schema.Resource{},
+		ConfigureFunc:  ConfigureProvider,
+	}
+}
+
+func ConfigureProvider(s *schema.ResourceData) (interface{}, error) {
+	dsn, err := DSN(s)
+
+	log.Printf("[DEBUG] connecting to %s", dsn)
```

Comment:
why not logrus.Debug?

Reply:
That would work if I can get the format right (has to start with "[DEBUG]" for terraform to pass it through correctly.

Reply:
I think I will leave it as-is for now.

Reply:
ah, interesting